src/snowflake/cli/_plugins/streamlit/manager.py

```diff
@@ -54,26 +59,46 @@ def share(self, streamlit_name: FQN, to_role: str) -> SnowflakeCursor:
 
     def _put_streamlit_files(
         self,
-        root_location: str,
-        artifacts: Optional[List[Path]] = None,
+        streamlit_project_paths: StreamlitProjectPaths,
+        stage_root: str,
+        artifacts: Optional[List[PathMapping]] = None,
     ):
-        cli_console.step(f"Deploying files to {root_location}")
+        cli_console.step(f"Deploying files to {stage_root}")
         if not artifacts:
             return
         stage_manager = StageManager()
-        for file in artifacts:
-            if file.is_dir():
-                if not any(file.iterdir()):
-                    cli_console.warning(f"Skipping empty directory: {file}")
-                    continue
+        bundle_map = BundleMap(
+            project_root=streamlit_project_paths.project_root,
+            deploy_root=streamlit_project_paths.deploy_root,
+        )
+        for artifact in artifacts:
+            bundle_map.add(PathMapping(src=artifact.src, dest=artifact.dest))
 
+        # Clean up deploy root
+        streamlit_project_paths.remove_up_deploy_root()
+
+        for (absolute_src, absolute_dest) in bundle_map.all_mappings(
+            absolute=True, expand_directories=True
+        ):
+            if absolute_src.is_file():
+                symlink_or_copy(
+                    absolute_src,
+                    absolute_dest,
+                    deploy_root=streamlit_project_paths.deploy_root,
+                )
+                # Temporary solution, will be replaced with diff
+                stage_path = (
+                    PurePosixPath(absolute_dest)
+                    .relative_to(streamlit_project_paths.deploy_root)
+                    .parent
+                )
+                stage_path = (
+                    stage_path if stage_path != PurePosixPath() else PurePosixPath()
```

Comment:
Discussed in person:
* `PurePosixPath()` returns current dir
* It looks like `else` does not do anything (`else` = if stage path is current dir, then set stage path to current dir)
* @sfc-gh-astus will check it, maybe it's required because of relative paths or sth like that

Reply:
Looks `if` is not needed, removed.